pkg/handlers/internalapi/weight_ticket.go

```
@@ -0,0 +1,142 @@
+	weightticketops "github.com/transcom/mymove/pkg/gen/internalapi/internaloperations/ppm"
+	"github.com/transcom/mymove/pkg/handlers"
+	"github.com/transcom/mymove/pkg/handlers/internalapi/internal/payloads"
+	"github.com/transcom/mymove/pkg/services"
+)
+
+// CreateWeightTicketHandler
+type CreateWeightTicketHandler struct {
+	handlers.HandlerConfig
+	weightTicketCreator services.WeightTicketCreator
+}
+
+// Handle creates a weight ticket
+// Depending on the SO, may need to change the document params to weight ticket params
+func (h CreateWeightTicketHandler) Handle(params weightticketops.CreateWeightTicketParams) middleware.Responder {
+	return h.AuditableAppContextFromRequestWithErrors(params.HTTPRequest,
+		func(appCtx appcontext.AppContext) (middleware.Responder, error) {
+			if appCtx.Session() == nil {
+				noSessionErr := apperror.NewSessionError("No user session")
+				return weightticketops.NewCreateWeightTicketUnauthorized(), noSessionErr
+			}
+
+			if !appCtx.Session().IsMilApp() && appCtx.Session().ServiceMemberID == uuid.Nil {
+				noServiceMemberIDErr := apperror.NewSessionError("No service member ID")
+				return weightticketops.NewCreateWeightTicketForbidden(), noServiceMemberIDErr
+			}
+
+			// NO NEED FOR payload_to_model, will need for Update
+			ppmShipmentID, err := uuid.FromString(params.PpmShipmentID.String())
+			if err != nil {
+				appCtx.Logger().Error("missing PPM Shipment ID", zap.Error(err))
+				return weightticketops.NewCreateWeightTicketBadRequest(), nil
+			}
+
+			weightTicket, err := h.weightTicketCreator.CreateWeightTicket(appCtx, ppmShipmentID)
+
+			if err != nil {
+				appCtx.Logger().Error("internalapi.CreateWeightTicketHandler", zap.Error(err))
+				// Can get a status error
+				// Can get an DB error - does the weight ticket, doc create?
+				// Can get an error for whether the PPM exist
+				switch err.(type) {
+				case apperror.InvalidInputError:
+					return weightticketops.NewCreateWeightTicketUnauthorized(), err
```

`apperror.InvalidInputError` should return an UnprocessableEntity error. There's some documentation on this https://github.com/transcom/mymove-docs/blob/main/docs/api/guides/api-errors.md#error-types-and-error-responses